src/LinkDotNet.NCronJob/Execution/JobExecutor.cs

```diff
@@ -51,13 +55,13 @@ public async Task RunJob(RegistryEntry run, CancellationToken stoppingToken)
 
         // stoppingToken is never cancelled when the job is triggered outside the BackgroundProcess,
         // so we need to tie into the IHostApplicationLifetime
-        using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(shutdown.Token, stoppingToken);
+        using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(shutdown.Token, stoppingToken, run.CancellationToken);
```

Comment:
The use of `CancellationToken` in job definitions could be ambiguous, especially in job chaining scenarios. There are two main cancellation scenarios to consider:

1. **Cancellation of a Single Job**: The `CancellationToken` should only affect the job it's passed to, ensuring targeted control.
2. **Cancellation of the Entire Job Chain**: In cases where there is a need to terminate the entire sequence of chained jobs, the `CancellationToken` should propagate across the entire chain to halt all subsequent jobs effectively.

Perhaps introducing two different cancellation tokens or a cancellation policy that can be specified at the job or job flow level would provide clearer control.

Reply:
That might be a good idea.
i do think I will take some off the good parts and split up this PR.